reconcile/template_tester.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import sys
+from typing import Any
+import yaml
+from reconcile.openshift_namespace_labels import label
+import reconcile.openshift_resources_base as orb
+
+from textwrap import indent
+from reconcile.status import ExitCodes
+from reconcile.utils import gql
+from reconcile.utils.semver_helper import make_semver
+from reconcile.utils.structs import DictObj
+
+
+QONTRACT_INTEGRATION = "template-tester"
+QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
+
+
+TEMPLATE_TESTS_QUERY = """
+{
+  tests: template_tests_v1 {
+    name
+    resource {
+      %s
+    }
+    resourceParent
+    testData
+    expectedResult
+  }
+}
+""" % indent(
+    orb.OPENSHIFT_RESOURCE, 2 * " "
+)
+
+
+def load_resource(path: str) -> dict:
+    return yaml.safe_load(gql.get_resource(path)["content"])
+
+
+def run(dry_run):
+    gqlapi = gql.get_api()
+    template_tests = gqlapi.query(TEMPLATE_TESTS_QUERY)["tests"]
+    error = False
+    for tt in template_tests:
+        gqlapi.init_test_data(load_resource(tt["testData"]))
+        resource = tt["resource"]
+        resource["add_path_to_prom_rules"] = False
+        parent = DictObj(load_resource(tt["resourceParent"]))
```

Comment:
what is the reason for the DictObj?

Reply:
no longer relevant with the new approach